permissions/Email.permission

```diff
@@ -19,7 +19,9 @@ mkdir     ${HOME}/Downloads/mail_attachments
 whitelist ${HOME}/Downloads/mail_attachments
 
 # MessageServer socket
-whitelist /tmp/qcop-server-0
+mkdir       ${RUNUSER}/messageserver
```

Comment:
Have you tested whether it works without doubling the mkdir?
I tend to think that it should. IIRC each mkdir is executed twice: first during parsing and again during sandbox setup along with white/blaclist etc -> even if Base is read last, 1st pass of mkdir in it ought to have chance to do its thing before rules from Email are acted upon.

Reply:
The thing I'm concerned is that if the blacklisted directory did not exists before the sandbox was constructed, the directory will become accessible when it's created. I tested this now and that seems to be the case. Since messageserver5 is only started on demand and the first sandbox constructed is voicecall-ui, then that means in practice it's always voicecall-ui sandboxing that's also creating it. I guess the alternative might be that we'd construct the directory when user session is started and then we wouldn't need any mkdir rules.

Reply:
Trying to clarify: IIRC each "mkdir" and "mkfile" op is acted on twice. First during command line arg / config file processing - when firejail is still acting as more or less regular user process in global namespace - and later on also from within sandboxing context -> even if a "mkdir" option is the last thing given to firejail, an attempt is made to create that directory regardless of other rules. And having the same "mkdir"  twice in config files means firejail tries to create that directory four times -> maybe it does not accumulate to much, but it is still excess when/if having it in one place (Base.permission?) would suffice.

Reply:
Ah, yes. Now I see what you mean. I did some quick testing and yes, having mkdir only in Base.permission suffices.